modules/python/nitf/source/nitro.i

```diff
@@ -570,15 +570,19 @@
         newWindow->numRows = window->numRows;
         newWindow->numCols = window->numCols;
         newWindow->numBands = 1;
-        
+
         result = PyList_New(window->numBands);
         for (i = 0; i < window->numBands; i++)
         {
             PyObject* buffObj = PyBuffer_New(subimageSize * sizeof(nitf_Uint8));
             if (!buffObj) goto CATCH_ERROR;
             buffObj->ob_type->tp_as_buffer->bf_getwritebuffer(buffObj, 0, (void **)&buf[0]);
-            
+
             newWindow->bandList = &window->bandList[i];
+
+            //We need to renumber the band going in to make a check pass,
+            //since we tell it that numBands = 1
+            newWindow->bandList[0] = 0;
```

Comment:
I don't understand why this is needed... if you hardcode `bandList` to always be 0, won't you not read out the band you want?  Seems like you'll read the same band each time through the loop.  Or am I missing something?